### ISSUE TITLE
update project's resources using curl command cause error  "Failed to update resources for project Unable to move temp file to dest file'

### ISSUE THREAD

**Issue by herrygoon** [rundeck/rundeck#1323]
curl command:

```
curl -v --header "X-Rundeck-Auth-Token:someToken" -H "Content-Type:text/xml" -d @"C:\Users\GSP\Desktop\resources.xml"  http://localhost:4440/api/13/project/testProj/resources
```

Exception:

```
2015-07-08 19:33:48,335 WARN  UpdateUtils - Unable to set modification time of temp file: E:\work\javaopensrc\rundeck\rundeck-launcher\projects\testProj\etc\resources.xml.new
2015-07-08 19:33:48,359 ERROR FrameworkController - Failed updating nodes file: Unable to move temp file to dest file: E:\work\javaopensrc\rundeck\rundeck-launcher\projects\testProj\etc\resources.xml.
new, E:\work\javaopensrc\rundeck\rundeck-launcher\projects\testProj\etc\resources.xml
com.dtolabs.rundeck.core.common.UpdateUtils$UpdateException: Unable to move temp file to dest file: E:\work\javaopensrc\rundeck\rundeck-launcher\projects\testProj\etc\resources.xml.new, E:\work\javaop
ensrc\rundeck\rundeck-launcher\projects\testProj\etc\resources.xml
        at com.dtolabs.rundeck.core.common.UpdateUtils.moveFile(UpdateUtils.java:117)
        at com.dtolabs.rundeck.core.common.UpdateUtils.update(UpdateUtils.java:210)
        at com.dtolabs.rundeck.core.common.UpdateUtils.updateFileFromInputStream(UpdateUtils.java:96)
        at com.dtolabs.rundeck.core.common.UpdateUtils.updateFileFromFile(UpdateUtils.java:81)
        at com.dtolabs.rundeck.core.common.ProjectNodeSupport.updateNodesResourceFile(ProjectNodeSupport.java:560)
        at com.dtolabs.rundeck.core.common.ProjectNodeSupport.updateNodesResourceFile(ProjectNodeSupport.java:545)
        at com.dtolabs.rundeck.core.common.FrameworkProject.updateNodesResourceFile(FrameworkProject.java:421)
```

And I found that the Class FileStreamUpdater's method updateFile in UpdateUtils don't close the input and output stream Cause the Exception.

```
private static class FileStreamUpdater implements FileUpdater {
        InputStream input;

        public FileStreamUpdater(InputStream input) {
            this.input = input;
        }

        public void updateFile(final File destinationFile) throws FileUpdaterException {
            FileOutputStream fos = null;
            try {
                try {
                    fos = new FileOutputStream(destinationFile);
                    Streams.copyStream(input, fos);
                } catch (IOException e) {
                    throw new FileUpdaterException(e);
                } finally {
                    //need to close the stream
                    if (input != null)
                        input.close();
                    if (fos != null)
                        fos.close();
                }
            } catch (IOException e) {
                throw new FileUpdaterException(e);
            }
        }
    }
```

**Comment by tclydesdale** [comment]
I just ran into this problem too. This is a pretty important feature that's broken, and seemingly very easy to fix.

Use case: Our CD pipeline spins up brand new VM's and I would like to automate adding and removing these VM's (nodes) to/from my projects, but this bug is preventing that.

Anyway, as far as a work around, I spent my morning learning how to compile a Java application! But obviously I don't want to have to do this for each release.

Thanks in advance!
